dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
 
-),
+- Use trailing commas
 
-some_cte_agg as (
+- Indents should be four spaces 
 
-    select
-        id,
-        sum(field_4) as total_field_4,
-        max(field_5) as max_field_5
+- When dealing with long `when` or `where` clauses, predicates should be on a new
+  line and indented.
 
-    from some_cte
-    group by 1
+- Lines of SQL should be no longer than 80 characters and new lines should be used to ensure this.  
+  Example:
+  ```sql
+  sum(
+    case when order_status = 'complete' then order_total end
+  ) as monthly_total,
 
-),
 
-final as (
+  {{ get_windowed_values(
+        strategy='sum',
+        partition='order_id',
+        order_by='created_at',
+        column_list=[
+          'final_cost'
+        ]
+  ) }} as total_final_cost
+  ```
 
-    select [distinct]
-        my_data.field_1,
-        my_data.field_2,
-        my_data.field_3,
+- Field names and function names should all be lowercase
 
-        -- use line breaks to visually separate calculations into blocks
-        case
-            when my_data.cancellation_date is null
-                and my_data.expiration_date is not null
-                then expiration_date
-            when my_data.cancellation_date is null
-                then my_data.start_date + 7
-            else my_data.cancellation_date
-        end as cancellation_date,
+- The `as` keyword should be used when aliasing a field or table
 
-        some_cte_agg.total_field_4,
-        some_cte_agg.max_field_5
+- Fields should be stated before aggregates / window functions
 
-    from my_data
-    left join some_cte_agg  
-        on my_data.id = some_cte_agg.id
-    where my_data.field_1 = 'abc'
-        and (
-            my_data.field_2 = 'def' or
-            my_data.field_2 = 'ghi'
-        )
-    having count(*) > 1
+- Aggregations should be executed as early as possible before joining to another table.
 
-)
+- Ordering and grouping by a number (eg. group by 1, 2) is preferred over listing the column names (see [this rant](https://blog.getdbt.com/write-better-sql-a-defense-of-group-by-1/) for why). Note that if you are grouping by more than a few columns, it may be worth revisiting your model design.
 
-select * from final
+- Prefer `union all` to `union` [*](http://docs.aws.amazon.com/redshift/latest/dg/c_example_unionall_query.html)
 
-```
+- Avoid table aliases in join conditions (especially initialisms) – it's harder to understand what the table called "c" is compared to "customers".
+
+- If joining two or more tables, _always_ prefix your column names with the table alias. If only selecting from one table, prefixes are not needed.
 
-- Your join should list the "left" table first (i.e. the table you are selecting `from`):
-```sql
-select
+- Be explicit about your join (i.e. write `inner join` instead of `join`). `left joins` are normally the most useful, `right joins` often indicate that you should change which table you select `from` and which one you `join` to.

```

Comment:
```suggestion
- Be explicit about your join (i.e. write `inner join` instead of `join`). `left joins` are normally the most common, `right joins` often indicate that you should change which table you select `from` and which one you `join` to.

```

Reply:
[Done](https://github.com/dbt-labs/corp/pull/63/commits/4b7f51a09054da94c9050527023b01d42f3e2e9e)